### ISSUE TITLE
Dataset removal not to kill Jena

### ISSUE THREAD

**Issue by jachro** [SwissDataScienceCenter/renku-graph#390]
It looks that dataset removal process still causes Jena to consume too many resources and so it's very slow.

**Acceptance criteria:**
- [x] the query checking if there are some triples to be deleted not to do a count but simply select a single non `CLIVersion` triple;
- [x] verify if the limit on the delete query works and maybe the solution from the point above could work better for the delete query, too;
- [x] add a semaphore to the `POST /events` (TG instances can be using some special triple in the Triples Store for synchronization) so:
  - [x] TG starting the re-provisioning process to set the special triple so it can be used by the other instances;
  - [x] TG finishing the re-provisioning process to clear the special triple once the process is finished;
  - [x] it responds with [`503 Service Unavailable`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/503) as long as the removal process is ongoing
  - [x] returns back to the operational state once the removal is finished;
  - [x] adapt event-log subscription mechanism so the `503`s cause putting a subscriber on hold;
  - [x] TG controlling re-provisioning to notify event-log about availability when the process is done;
  - [x] TG not controlling the re-provisioning process to notify event-log about availability when the flag is cleared;
  - [x] state of the re-provisioning flag to be cached;
- [x] use the CliVersion JSON object props in all the places;
- [x] remove the initial delay;
- [x] make a bugfix release.
